src/sparkbench/src/main/java/org/apache/spark/examples/mllib/JavaKMeans.java

```diff
@@ -87,7 +75,28 @@ public Vector call(Tuple2<LongWritable, VectorWritable> e) {
         }
     });
 
-    KMeansModel model = KMeans.train(points.rdd(), k, iterations, runs, KMeans.K_MEANS_PARALLEL());
+    // Load initial centroids
+    JavaPairRDD<Text, Kluster> clusters = sc.sequenceFile(inputCluster, Text.class, Kluster.class);
+    JavaRDD<Vector> centroids = clusters.map(new Function<Tuple2<Text, Kluster>, Vector>() {
+      @Override
+      public Vector call(Tuple2<Text, Kluster> e) {
+        org.apache.mahout.math.Vector centroid = e._2().getCenter();
+        double[] v = new double[centroid.size()];
+        for (int i = 0; i < centroid.size(); ++i) {
+          v[i] = centroid.get(i);
+        }
+        return Vectors.dense(v);
+      }
+    });
+
+    // Train model
+    KMeansModel initModel = new KMeansModel(centroids.collect());
+    KMeansModel model = new KMeans()
+        .setK(k)
+        .setMaxIterations(iterations)
+        .setRuns(runs)
+        .setInitialModel(initModel)
```

Comment:
I did the performance test on scala version Kmeans. The size of input data is about 100 GB across 4 nodes. The running time on `Random` and `Parallel` is almost same, which took about **4 mins** for running 3 iterations including **1.3 mins** on centroid initialization. So there is about **48.1%** overhead when using either the Spark version of `Random` or `Parallel`.  As a comparison, the implementation of this PR only took about **2.4 mins** for 3 iterations with almost zero-overhead on initialization.